lib/kazan/watcher.ex

```diff
@@ -220,32 +253,59 @@ defmodule Kazan.Watcher do
     %State{state | id: id, buffer: LineBuffer.new()}
   end
 
-  defp process_lines(
-         %State{name: name, send_to: send_to, rv: rv} = state,
-         lines
-       ) do
+  defp process_lines(%State{name: name, rv: rv} = state, lines) do
     debug(state, "Process lines: #{name}")
 
-    Enum.reduce(lines, rv, fn line, current_rv ->
-      {:ok, %{"type" => type, "object" => raw_object}} = Poison.decode(line)
-      {:ok, model} = Kazan.Models.decode(raw_object)
-
-      case {type, extract_rv(raw_object)} do
-        {_, ^current_rv} ->
-          Logger.warn(
-            "Duplicate message: #{name} type: #{type} rv: #{current_rv}"
-          )
-
-          current_rv
+    Enum.reduce(lines, {:ok, rv}, fn line, status ->
+      case status do
+        {:ok, current_rv} ->
+          process_line(line, current_rv, state)
 
-        {_, new_rv} ->
-          debug(state, "Received message: #{name} type: #{type} rv: #{new_rv}")
-          send(send_to, %WatchEvent{type: type, object: model})
-          new_rv
+        # stop on error
+        {:error, :gone} ->
+          {:error, :gone}
       end
     end)
   end
 
+  # Processes each line returning either {:ok, new_rv} or {:error, :gone}
+  # corresponding messages are sent to the send_to process
+  defp process_line(line, current_rv, %State{} = state) do
+    %State{name: name, send_to: send_to} = state
+    {:ok, %{"type" => type, "object" => raw_object}} = Poison.decode(line)
+    {:ok, model} = Kazan.Models.decode(raw_object)
+
+    case {type, extract_rv(raw_object)} do
```

Comment:
Maybe I'm missing something, but it doesn't look like we're using `type` in any of the matches.  Should we get rid of it from the case?